src/types/mod.rs

```diff
@@ -204,6 +205,60 @@ pub use self::capabilities::Capabilities;
 /// re-exported from imap_proto;
 pub use imap_proto::StatusAttribute;
 
+// We need a ResponseCode that is not tied to a lifetime, to be used in UnsolicitedResponse.
+/// Response code that may be sent with OK/NO/BAD/BYE responses.
+/// See [RFC 3501](https://tools.ietf.org/html/rfc3501#section-3.1).
+#[derive(Debug, Eq, PartialEq)]
+pub enum ResponseCode {
+    //Alert: not parsed by imap-proto yet.
+    //BadCharset: not parsed by imap-proto yet.
+    //Capability: not parsed by imap-proto yet.
+    //Parse: not parsed by imap-proto yet.
+    /// See [RFC 4551](https://tools.ietf.org/html/rfc4551#section-3.1.1).
+    HighestModSeq(u64),
+    /// Flags that can be changed permanently.
+    PermanentFlags(Vec<String>),
+    /// The mailbox status has changed to read-only.
+    ReadOnly,
+    /// The mailbox status has changed to read-write.
+    ReadWrite,
+    /// Indicates that the mailbox must be created first.
+    TryCreate,
+    /// Next unique identifier value.
+    UidNext(u32),
+    /// The unique identifier validity value.
+    UidValidity(u32),
+    /// First message without the \Seen flag set.
+    Unseen(u32),
+}
+
+impl<'a> From<imap_proto::types::ResponseCode<'a>> for ResponseCode {
```

Comment:
I wonder if [`ToOwned`](https://doc.rust-lang.org/std/borrow/trait.ToOwned.html) is the trait to implement here.